src/operator.rs

```diff
@@ -903,6 +905,62 @@ impl UnaryPreAggOperator for Total {
     }
 }
 
+pub struct SplitDef {
+    separator: String,
+    input_column: Option<String>,
+    output_column: String,
+}
+
+impl SplitDef {
+    pub fn new(separator: String, input_column: Option<String>, output_column: String) -> Self {
+        SplitDef {
+            separator,
+            input_column,
+            output_column,
+        }
+    }
+}
+
+impl OperatorBuilder for SplitDef {
+    fn build(&self) -> Box<dyn UnaryPreAggOperator> {
+        Box::new(Split::new(
+            self.separator.clone(),
+            self.input_column.clone(),
+            self.output_column.clone(),
+        ))
+    }
+}
+
+pub struct Split {
+    separator: String,
+    input_column: Option<Expr>,
+    output_column: String,
+}
+
+impl Split {
+    pub fn new(separator: String, input_column: Option<String>, output_column: String) -> Self {
```

Comment:
just take `input_column` as `: Option<Expr>` -- that ensures that everything works properly

Reply:
Done!